drivers/platform/mbed/mbed_uart.cpp

```diff
@@ -0,0 +1,492 @@
+{
+	_terminal_connected = connect_status;
+}
+
+/**
+ * @brief	Check if new USB data is received/available
+ * @param	bytes[in] - Number of expected bytes to be received
+ * @return	true if expected number of bytes received, else false
+ */
+bool platform_usbcdc::data_received(uint32_t bytes)
+{
+	volatile static uint32_t *rx_size = &_rx_size;
+
+	if (*rx_size >= bytes)
+		return true;
+	else
+		return false;
+}
+
+/**
+ * @brief  Check if old USB data was transmitted
+ * @return true if transmit not in progress, else false
+ */
+bool platform_usbcdc::data_transmited(void)
+{
+	volatile static bool *tx_in_progress = &_tx_in_progress;
+	return !(*tx_in_progress);
+}
+
+/**
+ * @brief Read data from UART device.
+ * @param desc[in, out] - Instance of UART.
+ * @param data[out] - Pointer to buffer containing data.
+ * @param bytes_number[in] - Number of bytes to read.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+int32_t uart_read(struct uart_desc *desc, uint8_t *data,
+		  uint32_t bytes_number)
+{
+	uint32_t cnt;
+	mbed::BufferedSerial *uart;	// pointer to BufferedSerial/UART instance
+	platform_usbcdc *usb_cdc_dev;	// Pointer to usb cdc device class instance
+	uint32_t size_rd;
+
+	if (!desc || !desc->extra || !data)
+		return -EINVAL;
+
+	if (!((struct mbed_uart_desc *)(desc->extra))->uart_port)
+		return -EINVAL;
+
+	if (((struct mbed_uart_desc *)desc->extra)->virtual_com_enable) {
+		usb_cdc_dev = (platform_usbcdc *)((struct mbed_uart_desc *)(
+				desc->extra))->uart_port;
+
+		while (!usb_cdc_dev->data_received(bytes_number)) {
+			/* Wait until new data is available */
+		}
+
+		/* Change terminal connection status manually */
+		usb_cdc_dev->change_terminal_connection(true);
+
+		usb_cdc_dev->receive_nb(data, bytes_number, &size_rd);
+	} else {
+		uart = (BufferedSerial *)(((struct mbed_uart_desc *)(
+						   desc->extra))->uart_port);
+
+		for (cnt = 0; cnt < bytes_number; cnt++)
+			uart->read(data + cnt, 1);
+	}
+
+	return bytes_number;
+}
+
+/**
+ * @brief Write data to UART device.
+ * @param desc[in] - Instance of UART.
+ * @param data[in, out] - Pointer to buffer containing data.
+ * @param bytes_number[in] - Number of bytes to read.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+int32_t uart_write(struct uart_desc *desc, const uint8_t *data,
+		   uint32_t bytes_number)
+{
+	mbed::BufferedSerial *uart;	// pointer to BufferedSerial/UART instance
+	platform_usbcdc *usb_cdc_dev;	// Pointer to usb cdc device class instance
+	uint32_t d_sz;
+	uint32_t indx = 0;
+	uint32_t tx_bytes = bytes_number;
+
+	if (!desc || !desc->extra || !data)
+		return -EINVAL;
+
+	if (!((struct mbed_uart_desc *)(desc->extra))->uart_port)
+		return -EINVAL;
+
+	if (((struct mbed_uart_desc *)desc->extra)->virtual_com_enable) {
+		usb_cdc_dev = (platform_usbcdc *)((struct mbed_uart_desc *)(
+				desc->extra))->uart_port;
+
+		while (bytes_number) {
```

Comment:
nitpick: `while (true)` is more correct, if what we want to do is wait for a previous transfer to complete. We never know where bugs might arise.